File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/wrapped/custom/wrappers/CarWrapper.java
package com.github.klee0kai.test.car.di.wrapped.custom.wrappers;

import com.github.klee0kai.stone.annotations.wrappers.WrappersCreator;
import com.github.klee0kai.stone.types.wrappers.Wrapper;

@WrappersCreator(wrappers = {CarRef.class})
public class CarWrapper implements Wrapper {

    @Override
    public <Wr, T> Wr wrap(Class<Wr> wrapperCl, T original) {
        if (wrapperCl.equals(CarRef.class)) {
            return (Wr) new CarRef<>(original);

        }
        return null;
    }
}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/wrapped/custom/wrappers/CarWrapper.kt
package com.github.klee0kai.test.car.di.wrapped.custom.wrappers

import com.github.klee0kai.stone.annotations.wrappers.WrappersCreator
import com.github.klee0kai.stone.wrappers.creators.Wrapper

@WrappersCreator(wrappers = [CarRef::class])
class CarWrapper : Wrapper {
    override fun <Wr, T> wrap(wrapperCl: Class<Wr>, original: T): Wr {
        return if (wrapperCl == CarRef::class.java) {
            CarRef(original) as Wr
        } else null as Wr
    }
}
